src/ripple/app/tx/apply.h

```diff
@@ -56,38 +65,57 @@ checkValidity(HashRouter& router,
 
 
 /** Sets the validity of a given transaction in the cache.
-    Use with extreme care.
+
+    @warn Use with extreme care.
 
     @note Can only raise the validity to a more valid state,
           and can not override anything cached bad.
+
+    @see checkValidity
 */
 void
 forceValidity(HashRouter& router, uint256 const& txid,
     Validity validity);
 
-/** Apply a transaction to a ReadView.
-
-    Throws:
-        Does not throw.
-
-        For open ledgers, the Transactor will catch and
-        return tefEXCEPTION. For closed ledgers, the
-        Transactor will attempt to only charge a fee,
-        and return tecFAILED_PROCESSING.
-
-        If the Transactor gets an exception while trying
-        to charge the fee, it will be caught here and
-        turned into tefEXCEPTION.
-
-        This try/catch handler is the last resort, any
-        uncaught exceptions will be turned into
-        tefEXCEPTION.
-
-        For network health, a Transactor makes its
-        best effort to at least charge a fee if the
-        ledger is closed.
-
-    @return A pair with the TER and a bool indicating
+/** Apply a transaction to an OpenView.
+
+    This function is the canonical way to apply a transaction
+    to a ledger. It rolls the validation and application
+    steps into one function. To do the steps manually, the
+    correct calling order is:
+    @code{.cpp}
+    preflight -> preclaim -> doApply
+    @endcode
+    And the result of one function must be passed to the next.
+    The `preflight` result can be safely cached and reused
+    asynchronously, but `preclaim` and `doApply` must be called
+    in the same thread and with the same view.
+
+    @throws none Does not throw.
```

Comment:
Yeah, I wasn't super happy with that either. Maybe I'll just do it as a `@note`...